File: Leetcode/typescript/884-uncommon-words.ts
function uncommonFromSentences(s1: string, s2: string): string[] {
  const words = s1.split(" ").concat(s2.split(" "));

  const wordCount = {};

  for (const word of words) {
    wordCount[word] = (wordCount[word] || 0) + 1;
  }
  let res: string[] = [];
  for (const word in wordCount) {
    if (wordCount[word] === 1) {
      res.push(word);
    }
  }
  return res;
}

File: Leetcode/python/884-uncommon-words.py
from typing import List
from collections import defaultdict


class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        count = defaultdict(int)

        for w in s1.split(' ') + s2.split(' '):
            count[w] += 1

        res = [
            w for w, cnt in count.items() if cnt == 1
        ]

        return res
